tests/strings/collapseSpaces.du

```diff
@@ -0,0 +1,19 @@
+/**
+ * collapseSpaces.du
+ *
+ * Testing the str.collapseSpaces() method
+ *
+ * .collapseSpaces() returns a string with extraneous spaces removed.
+ */
+from UnitTest import UnitTest;
+
+class TestStringCollapseSpaces < UnitTest {
+    testStringCollapseSpaces() {
+        const testString = "This  is      a huge string of a       lot of spaces.";
+        const expected  = "This is a huge string of a lot of spaces.";
+        const res = testString.collapseSpaces();
```

Comment:
Probably worth adding some tests to ensure the original string is also untouched